uaclient/clouds/tests/test_aws.py

```diff
@@ -17,6 +20,44 @@ def test_identity_doc_from_aws_url_pkcs7(self, readurl):
         url = "http://169.254.169.254/latest/dynamic/instance-identity/pkcs7"
         assert [mock.call(url)] == readurl.call_args_list
 
+    @pytest.mark.parametrize("caplog_text", [logging.DEBUG], indirect=True)
+    @pytest.mark.parametrize(
+        "fail_count,exception", ((3, None), (4, HTTPError))
+    )
+    @mock.patch(M_PATH + "util.time.sleep")
+    @mock.patch(M_PATH + "util.readurl")
+    def test_retry_backoff_on_failed_identity_doc(
+        self, readurl, sleep, fail_count, exception, caplog_text
+    ):
+        """Retry backoff is attempted before failing to get AWS.identity_doc"""
+
+        def fake_someurlerrors(url):
+            if readurl.call_count <= fail_count:
+                raise HTTPError(
+                    None, 400 + readurl.call_count, None, None, BytesIO()
+                )
+            return "pkcs7WOOT!==", {"header": "stuff"}
+
+        readurl.side_effect = fake_someurlerrors
+        instance = UAPremiumAWSInstance()
+        if exception:
+            with pytest.raises(HTTPError) as excinfo:
+                instance.identity_doc
+            assert 404 == excinfo.value.code
+        else:
+            assert "pkcs7WOOT!==" == instance.identity_doc
+
+        expected_sleep_calls = [mock.call(1), mock.call(2), mock.call(5)]
+        assert expected_sleep_calls == sleep.call_args_list
+        expected_logs = [
+            "HTTP Error 401: None Retrying 3 more times.",
```

Comment:
We pass `None` into `HTTPError`s constructor three times; should we pass an actual value for whichever of those this "None" is, to be sure that we're including the correct thing in these log messages?

Reply:
Changed to add the error message instead of None.